core/src/main/java/nl/nn/adapterframework/compression/ZipWriterPipe.java

```diff
@@ -38,18 +50,73 @@
  */
 public class ZipWriterPipe extends CollectorPipeBase<ZipWriter, MessageZipEntry> {
 
-	private @Getter boolean includeFileHeaders=false;
+	private @Getter boolean includeFileHeaders = false;
+
+	private boolean backwardsCompatibility = false;
+
+	public ZipWriterPipe() {
+		setCollectionName("zipwriterhandle");
+	}
 
 	@Override
 	public void configure() throws ConfigurationException {
-		setCollectionName("zipwriterhandle");
 		super.configure();
 		ZipWriter.validateParametersForAction(getAction(), getParameterList());
 	}
 
 	@Override
-	protected ZipWriter createCollector() throws CollectionException {
-		return new ZipWriter(includeFileHeaders);
+	protected ZipWriter createCollector(Message input, PipeLineSession session) throws CollectionException {
+		String filename = ParameterValueList.getValue(getParameterValueList(input, session), ZipWriter.PARAMETER_FILENAME, "");
+		if(backwardsCompatibility && input.asObject() instanceof String && StringUtils.isEmpty(filename)) {
+			try {
+				filename = input.asString();
+			} catch (IOException e) {
+				throw new CollectionException("unable to convert inputstring to filename");
+			}
+		}
+		return new ZipWriter(includeFileHeaders, filename);
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message input, PipeLineSession session) throws PipeRunException {
+		if(backwardsCompatibility) {
+			try {
+				if(Action.STREAM == getAction()) {
+					return handleStreamAction(input, session);
```

Comment:
Het ging hier fout omdat de zip in een andere zip moest komen. Daar was vroeger voor bedacht dat je een outputstream mee kon geven.